sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -53,6 +54,23 @@ def insert(
             for item in items
         ]
 
+        # Remove duplicates if they already exist
+        deduplicated_items = []
+        for item in items:
+            item_hash = utils.compute_content_hash(item)
+
+            if item_hash in self._hash_to_id:
+                if item.id is None or self._hash_to_id[item_hash] == item.id:  # type: ignore
+                    LOGGER.debug(
+                        "Duplicate item found with hash: %s - ignored the event",
+                        item_hash,
+                    )
+                    continue
+
+            deduplicated_items.append(item)
+            self._hash_to_id[item_hash] = item.id  # type: ignore
+            self._id_to_hash[item.id] = item_hash  # type: ignore
```

Comment:
I would suggest extracting this code into a separate method. This would slightly reduce the method's complexity (McCabe/Cyclomatic Complexity) and improve testability by allowing you to test just the deduplication logic without needing to mock the Opik client or other dependencies.